common/main/java/com/couchbase/lite/internal/core/C4CollectionObserver.java

```diff
@@ -101,9 +107,11 @@ private C4CollectionObserver(@NonNull NativeImpl impl, long token, long peer, @N
     // public methods
     //-------------------------------------------------------------------------
 
-    @NonNull
-    public C4DocumentChange[] getChanges(int maxChanges) {
-        return withPeerOrThrow((peer) -> impl.nGetChanges(peer, maxChanges));
+    @Override
+    @Nullable
+    public List<C4DocumentChange> getChanges(int maxChanges) {
+        final C4DocumentChange[] changes = withPeerOrThrow((peer) -> impl.nGetChanges(peer, maxChanges));
+        return (changes.length <= 0) ? null : Arrays.asList(changes);
```

Comment:
So it doesn't filter null items in changes anymore? Or does the null item get filtered somewhere else?

Reply:
I saw it.